### DISCUSSION TITLE
Tracking crypto-currency assets not in USD

### DISCUSSION THREAD

**Discussion by pcaro90** [ghostfolio/ghostfolio#2317] [Q&A] [answered]
Hello there,

Some versions ago, the recommended way to track crypto-currency assets using something different than USD was setting the "BASE_CURRENCY" environment variable (as seen in #945).

However, in Ghostfolio v2.0.0, that env variable is deprecated. Right now, I cannot add activity using "BTC-EUR" as symbol (even if Yahoo Finances actually tracks "BTC-EUR" and my Base Currency option being set to "EUR").

What's the best way to track BTC-EUR in v2.0.0?

Also, @dtslvr, thank you for an awesome tool; I hope this community keeps growing.

**Comment by dtslvr** [comment, marked as answer]
Hi @pcaro90

For some time now, there is a feature to set a custom currency during manual recording, where the unit price will be converted automatically. There is a similar functionality when importing activities.

**Reply by pcaro90** [reply]
I see, so the recommended way is adding the operation using EUR (or any other currency) and let Ghostfolio to convert it to USD, even if the purchase was done in EUR; like this:

![image](https://github.com/ghostfolio/ghostfolio/assets/1571486/e22ae6b8-cc1d-4361-9bdd-4e1296edb57b)

I think it's kind of confusing tracking BTC-USD using converted EUR, instead of just using BTC-EUR. For example, the operation total (21.395,64 USD, marked in green in my example) doesn't match the actually spent amount (20.000 EUR), which makes it difficult to compare against my broker records.

Crypto-coins are not a large share of my portfolio, so this is fine for me; but I can see this being a problem for people with different allocations.

**Reply by MRobi1** [reply]
This is absolutely a problem. BTC is purchased in local currency (in my case CAD) from domestic exchanges. At no point in the transaction is USD being used. Forcing a conversion to USD based on historical data for that specific date will not provide accurate record keeping for tax reporting purposes as currency exchange fluctuates throughout the day. Also, all activities are showing in USD when USD was never used in the transaction. 

@dtslvr is there a way to avoid the forced currency conversion? And is there a way to easily convert the hundreds of transactions away from Yahoo without manually recreating every single one of them? Or could we just have support for Yahoo crypto re-added since it appears to still be used for everything else?

**Reply by pcaro90** [reply]
I ended up manually adding BTC-EUR as symbol (see #2295), which seems to work perfectly fine (while it feels kinda "hackish").

**Reply by MRobi1** [reply]
From what I'm understanding there, that involves creating a new database entry for the symbol, and then all new activities need to also be added through modifying the database. And pricing tracked manually?

**Reply by pcaro90** [reply]
It does involves creating a new database entry. However, as [BTC-CAD](https://finance.yahoo.com/quote/BTC-CAD) actually exists in Yahoo Finance, price tracking is done automatically. You can use that same symbol to add new transactions as normal, without further interacting with the database.

However, for existing transactions using BTC instead of BTC-CAD, I assume you'll need some kind of conversion.

**Reply by MRobi1** [reply]
```
ghostfolio-db=# INSERT INTO "SymbolProfile" ("name", "symbol", "currency", "dataSource", "id", "updatedAt") VALUES ('Bitcoin CAD', 'BTCCAD', 'CAD', 'YAHOO', '4f79da65-cb33-400c-8d2b-53e288af5042', current_timestamp);
ERROR:  duplicate key value violates unique constraint "SymbolProfile_dataSource_symbol_key"
DETAIL:  Key ("dataSource", symbol)=(YAHOO, BTCCAD) already exists.
```
So I'd say the entry already exists, it's likely just no longer pointed to Yahoo. Any idea how to modify the existing entry?

EDIT: I can get the new entry added by making it BTC-CAD instead of BTCCAD. And I can get the market history imported from YAHOO without issues. But I can not add any new activities using it because it doesn't validate in the name/symbol entry field.

![image](https://github.com/ghostfolio/ghostfolio/assets/34860474/05f9759b-5403-4969-a059-760439a11966)

**Comment by dtslvr** [comment]
Here is a crowdfunding campaign to further improve the support for custom currencies: https://www.buymeacoffee.com/ghostfolio/w/31930

**Comment by MRobi1** [comment]
I opened up a new issue [here](https://github.com/ghostfolio/ghostfolio/issues/2700) because attempting to do this using the new data provider in USD with forced currency conversion gives wildly inaccurate results. Either the new crypto implementation is broken, or the currency conversion is broken. But this does not work.
